src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.order.CreateOrderRequestDto;
+import mate.academy.dto.order.OrderDto;
+import mate.academy.dto.order.UpdateOrderDto;
+import mate.academy.dto.order.UpdateOrderResponseDto;
+import mate.academy.dto.orderitem.OrderItemDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.exception.OrderProcessingException;
+import mate.academy.mapper.OrderItemMapper;
+import mate.academy.mapper.OrderMapper;
+import mate.academy.model.Order;
+import mate.academy.model.Order.Status;
+import mate.academy.model.OrderItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.repository.OrderItemRepository;
+import mate.academy.repository.OrderRepository;
+import mate.academy.repository.ShoppingCartRepository;
+import mate.academy.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository cartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository itemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(Long userId, CreateOrderRequestDto orderDto) {
+        ShoppingCart cart = cartRepository.findByUserId(userId);
+        if (cart.getCartItems().isEmpty()) {
+            throw new OrderProcessingException("Cart is empty for user: " + userId);
+        }
+        Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());
+        cart.clearCart();
+        return orderMapper.toOrderDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderDto> getAllOrders(Long userId, Pageable pageable) {
+        return orderMapper.toOrderDtoList(orderRepository.findAllByUserId(userId, pageable));
+    }
+
+    @Override
+    public UpdateOrderResponseDto updateOrderStatus(Long orderId, UpdateOrderDto orderDto) {
+        Order order = orderRepository.findById(orderId)
+                .map(o -> {
+                    o.setStatus(Status.valueOf(orderDto.status()));
+                    return o;
+                })
```

Comment:
I think you can remove this map method and simly set the updated status after retrieving order

this will improve readability